lib/motors/vector_controller.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <cmath>
+
+#include "vector_controller.hpp"
+
+namespace hyped :: motors {
+    
+    VectorController::VectorController() {
+        Kp = 1.0;  
+        Ki = 0.1; 
+        IdRef = 0.0;  
+        IqRef = 0.0;  
+        theta = 0.0; 
+    }
+
+            // Measure stator currents and rotor velocity
+            void VectorController::measureCurrentsAndVelocity(core::Float ia, core::Float ib, core::Float ic, core::Float velocity) {
+                Ia = ia;
+                Ib = ib;
+                Ic = ic;
+                rotorVelocity = velocity;
+            }
+
+            // Convert to 2-axis system
+            void VectorController::convertTo2Axis() {
```

Comment:
these functions should take parameters and return values instead of doing random stuff to internal variables, cos until the end of this function there will be a `Id` and `Iq` which are outdated